modules/extensions/src/collide/collide-effect.ts

```diff
@@ -0,0 +1,252 @@
+      views: PreRenderOptions['views'];
+      viewport: Viewport;
+      viewportChanged: boolean;
+    }
+  ) {
+    const {collideGroup} = renderInfo;
+    const oldRenderInfo = this.channels[collideGroup];
+    if (!oldRenderInfo) {
+      return;
+    }
+
+    const renderInfoUpdated =
+      // If render info is new
+      renderInfo === oldRenderInfo ||
+      // If sublayers have changed
+      oldRenderInfo.layers.length !== renderInfo.layers.length ||
+      // If a sublayer's bounds have been updated
+      renderInfo.layerBounds.some((b, i) => !equals(b, oldRenderInfo.layerBounds[i]));
+
+    this.channels[collideGroup] = renderInfo;
+
+    if (renderInfoUpdated || viewportChanged) {
+      this.lastViewport = viewport;
+      const collideFBO = this.collideFBOs[collideGroup];
+      const collidePass = this.collidePasses[collideGroup];
+
+      // Rerender collide FBO
+      // @ts-ignore (2532) This method is only called from preRender where collidePass is defined
+      collidePass.renderCollideMap(collideFBO, {
+        pass: 'collide',
+        layers: renderInfo.layers,
+        effects,
+        layerFilter,
+        viewports: viewport ? [viewport] : [],
+        onViewportActive,
+        views,
+        moduleParameters: {
+          // To avoid feedback loop forming between Framebuffer and active Texture.
+          dummyCollideMap: this.dummyCollideMap,
+          devicePixelRatio: cssToDeviceRatio(collideFBO.gl) / DOWNSCALE
+        }
+      });
+    }
+  }
+
+  /**
+   * Group layers by collideGroup
+   * Returns a map from collideGroup to render info
+   */
+  private _groupByCollideGroup(
+    gl: WebGLRenderingContext,
+    collideLayers: Layer<CollideExtensionProps>[]
+  ): Record<string, RenderInfo> {
+    const channelMap = {};
+    for (const layer of collideLayers) {
+      const {collideGroup} = layer.props;
+      let channelInfo = channelMap[collideGroup];
+      if (!channelInfo) {
+        channelInfo = {
+          collideGroup,
+          layers: [],
+          layerBounds: []
+        };
+        channelMap[collideGroup] = channelInfo;
+      }
+      channelInfo.layers.push(layer);
+      channelInfo.layerBounds.push(layer.getBounds());
+    }
+
+    // Create any new passes and remove any old ones
+    for (const collideGroup of Object.keys(channelMap)) {
+      if (!this.collidePasses[collideGroup]) {
+        this.createPass(gl, collideGroup);
+      }
+      if (!this.channels[collideGroup]) {
+        this.channels[collideGroup] = channelMap[collideGroup];
+      }
+    }
+    for (const collideGroup of Object.keys(this.collidePasses)) {
+      if (!channelMap[collideGroup]) {
+        this.destroyPass(collideGroup);
+      }
+    }
+
+    return channelMap;
+  }
+
+  getModuleParameters(): {collideFBOs: Record<string, Framebuffer>; dummyCollideMap: Texture2D} {
+    const {collideFBOs, dummyCollideMap} = this;
+    return {collideFBOs, dummyCollideMap};
```

Comment:
Just pass a single FBO by `collideFBO[layer.props.collideGroup]`. The less the layer knows about the internals of the effect, the better.